### PR TITLE
ext authz: add dns san support for ext authz service

### PR DESCRIPTION
**Pull request by ramaraochavali**
Description: Adds support for DNS SAN in ext authz peer validation
Risk Level: Low
Testing: Added
Docs Changes: Added
Release Notes: N/A

Signed-off-by: Rama Chavali <rama.rao@salesforce.com>

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@lizan Same as #7881 , but for ext authz. Can you PTAL?

**Comment by ramaraochavali** [comment]
@dio Can you PTAL when you get chance?

File: source/extensions/filters/common/ext_authz/check_request_utils.cc [hunk 1]
```diff
@@ -45,14 +45,24 @@ void CheckRequestUtils::setAttrContextPeer(envoy::service::auth::v2::AttributeCo
     if (local) {
       const auto uriSans = ssl->uriSanLocalCertificate();
       if (uriSans.empty()) {
-        peer.set_principal(ssl->subjectLocalCertificate());
```
[the remark this answers is not in the document]
Comment (ramaraochavali) [reply source/extensions/filters/common/ext_authz/check_request_utils.cc]: Update the comment.

**Comment by ramaraochavali** [comment]
@dio updated the comments in code and proto. PTAL.

**Comment by ramaraochavali** [comment]
@htuch can you please approve the api change?

File: source/extensions/filters/common/ext_authz/check_request_utils.cc [hunk 2]
```diff
@@ -37,22 +37,31 @@ void CheckRequestUtils::setAttrContextPeer(envoy::service::auth::v2::AttributeCo
...
-  // Subject from the peer's cert.
+  // Set the principal. Preferably the URI SAN, DNS SAN or Subject in that order from the peer's
+  // cert.
   Ssl::ConnectionInfo* ssl = const_cast<Ssl::ConnectionInfo*>(connection.ssl());
   if (ssl != nullptr) {
     if (local) {
       const auto uriSans = ssl->uriSanLocalCertificate();
       if (uriSans.empty()) {
-        peer.set_principal(ssl->subjectLocalCertificate());
+        const auto dnsSans = ssl->dnsSansLocalCertificate();
```
Comment (htuch) [inline source/extensions/filters/common/ext_authz/check_request_utils.cc]: Nit: `dns_sans`

File: source/extensions/filters/common/ext_authz/check_request_utils.cc [hunk 3]
```diff
@@ -37,22 +37,31 @@ void CheckRequestUtils::setAttrContextPeer(envoy::service::auth::v2::AttributeCo
...
 
-  // Set the principal
-  // Preferably the SAN from the peer's cert or
-  // Subject from the peer's cert.
+  // Set the principal. Preferably the URI SAN, DNS SAN or Subject in that order from the peer's
+  // cert.
   Ssl::ConnectionInfo* ssl = const_cast<Ssl::ConnectionInfo*>(connection.ssl());
   if (ssl != nullptr) {
     if (local) {
       const auto uriSans = ssl->uriSanLocalCertificate();
```
Comment (htuch) [inline source/extensions/filters/common/ext_authz/check_request_utils.cc]: Nit: already there, but `uri_sans`

File: test/extensions/filters/common/ext_authz/check_request_utils_test.cc [hunk 4]
```diff
@@ -127,45 +154,50 @@ TEST_F(CheckRequestUtilsTest, BasicHttpWithFullBody) {
...
+
   EXPECT_CALL(ssl_, uriSanPeerCertificate()).WillOnce(Return(std::vector<std::string>{"source"}));
   EXPECT_CALL(ssl_, uriSanLocalCertificate())
       .WillOnce(Return(std::vector<std::string>{"destination"}));
 
+  callHttpCheckAndValidateRequestAttributes();
+}
+
+// Verify that createHttpCheck extract the attributes from the http request into CheckRequest
+// proto object and Dns SAN is used as principal if Uri SAN is absent.
```
Comment (htuch) [inline test/extensions/filters/common/ext_authz/check_request_utils_test.cc]: Nit: s/Dns/DNS, s/http request/HTTP rquest, etc. here and elsewhere in comments. Same with Uri -> URI in comments.

**Comment by ramaraochavali** [comment]
@htuch addressed the feedback. PTAL

**Pull request merged**